onnxoptimizer/__init__.py

```diff
@@ -42,14 +45,33 @@ def optimize(model, passes=None, fixed_point=False):  # type: (ModelProto, Optio
     if not isinstance(model, ModelProto):
         raise ValueError(
             'Optimizer only accepts ModelProto, incorrect type: {}'.format(type(model)))
+    try:
+        model_str = model.SerializeToString()
+        if fixed_point:
+            optimized_model_str = C.optimize_fixedpoint(model_str, passes)
+        else:
+            optimized_model_str = C.optimize(model_str, passes)
 
-    model_str = model.SerializeToString()
-    if fixed_point:
-        optimized_model_str = C.optimize_fixedpoint(model_str, passes)
-    else:
-        optimized_model_str = C.optimize(model_str, passes)
-
-    return onnx.load_from_string(optimized_model_str)
+        return onnx.load_from_string(optimized_model_str)
+    except ValueError:
+        file_src = tempfile.NamedTemporaryFile(suffix=".onnx", delete=False)
+        file_dest = tempfile.NamedTemporaryFile(suffix=".onnx", delete=False)
+        data_file_src = tempfile.NamedTemporaryFile(delete=False)
+        data_file_dest = tempfile.NamedTemporaryFile(delete=False)
+        data_src_rel_filename = os.path.relpath(data_file_src.name, file_src.name)
+        data_dest_rel_filename = os.path.relpath(data_file_dest.name, file_dest.name)
+        try:
+            onnx.save(model, file_src, save_as_external_data=True, location=data_src_rel_filename, convert_attribute=True,)
+            if fixed_point:
+                C.optimize_fixedpoint_from_path(file_src.name, file_dest.name, passes, data_dest_rel_filename)
+            else:
+                C.optimize_from_path(file_src.name, file_dest.name, passes, data_dest_rel_filename)
+            return onnx.load(file_dest, load_external_data=True)
+        finally:
+            os.remove(file_src.name)
+            os.remove(file_dest.name)
+            os.remove(data_file_src.name)
+            os.remove(data_file_dest.name)
```

Comment:
why not `delete=True` and close them instead of explicitly delete them

Reply:
https://docs.python.org/3/library/tempfile.html  `tempfile.NamedTemporaryFile  ` mentioned:
```
Whether the name can be used to open the file a second time, 
while the named temporary file is still open, 
varies across platforms (it can be so used on Unix; it cannot on Windows). 
```
I think it is better to delete temporary file manually than to remove by system .